safe-global/safe-wallet-web
# Feat: EIP-5792 (tx bundling) for WalletConnect

## What it solves

This PR adds support for [EIP-5792](https://eips.ethereum.org/EIPS/eip-5792), so that dapps connected to Safe via WalletConnect can bundle several transactions in one RPC call. This gets translated to a multisend call on our side.

Apart from bundling itself, this API is very good for us (and Smart Accounts in general) because it decouples transaction sending and transaction status retrieval. A regular `eth_sendTransaction` expects a tx hash in the response, which we can only provide in 1/X Safes.

With this EIP, the call itself is made via `wallet_sendFunctionCallBundle` which returns our offchain safeTxHash, and a dapp can monitor the tx status via another call, `wallet_getBundleStatus`.

## How to test it

It can be tested with this test dapp: https://github.com/5afe/eip-1271-dapp
(Press "Send", then "Check status")
